releasenotes/notes/add-call-instruction-6f4fc37994090c2f.yaml

```diff
@@ -13,26 +13,41 @@ deprecations:
 features:
   - |
     Pulse :py:class:`~qiskit.pulse.instructions.Call` instruction is added.
-    This instruction is a compiler directive and not exposed to end users.
-    The logic of the pulse builder is updated to store a call instruction when py:func:`pulse.call`
-    function is used with ``target``, i.e. ``QuantumCircuit`` or ``Schedule``.
-    Now this function saves the ``target`` as a call instruction rather than simply appending
-    its schedule to the context. This representation explicitly describes
-    the ``target`` is defined outside of the current scope, or as a subroutine.
-    No further schedule transforms are applied between subroutines and stored programs
-    are kept as-is. For example,
+    This instruction represents calling a schedule as a subroutine from within another schedule.
+    This is not exposed to end users, but internally used for compiling purpose.
```

Comment:
Is this really not exposed to end-users? It seems like any user of a Schedule could use this straightforwardly?

Reply:
Are we sure we do not want to expose this to the user? It could be very useful in the calibration module for instance.

Reply:
I think this depends on how we define _expose_. I don't think a developer of calibration module is not an end user. Because Call instruction doesn't trigger any operation on the Schedule so user should not aware of its existence. This is just a compiler issue and we can run the same program without Call. I'm not sure if we really need to ask end users to program the pulse schedule by considering compiler stuff, i.e.

I think this is quite easy to understand
```python
xsched = Schedule()
xsched += Play(Drag(160, 0.1, 40, 0.5), DriveChannel(0))

sched = Schedule()
sched += xsched
sched += Play(Drag(160, 0.2, 40, 0.5), DriveChannel(0))
```

but I think below example is very confusing if a user is not a software engineer (though Call is quite popular instruction which can be seen in many programming languages):
```python
sched = Schedule()
sched += Call(xsched)
sched += Play(Drag(160, 0.2, 40, 0.5), DriveChannel(0))  # why this is not Call ed? - of course it can be
```

Note that above examples will generate identical pulse sequence, and thus user may confuse when to insert the Call instruction. Actually the user can use Call for every `+=` sugar, but this will just mess up the target code.

In the pulse builder this is more intuitive because it has notion of context
```python
with pulse.build() as sched:
    pulse.call(xsched)
    pulse.play(Drag(160, 0.2, 40, 0.5), DriveChannel(0))
```
It can be understood Call is applied if we reuse schedules defined outside of the current context.

Reply:
I think this is one of reasons we want to discourage direct use of Schedule.